src/dwi/tractography/resampling/arc.cpp

```diff
@@ -64,16 +63,19 @@ namespace MR {
           size_t i = idx_start;
 
           for (size_t n = 0; n < nsamples; n++) {
-            while (i != idx_end) {
+            do {
               const value_type d = planes[n].dist (in[i]);
               if (d > 0.0) {
                 const value_type f = d / (d - planes[n].dist (in[reverse ? i+1 : i-1]));
-                out.push_back (f*in[i-1] + (1.0f-f)*in[i]);
+                assert (f >= 0.0 && f <= 1.0);
+                out.push_back (f*in[reverse ? i+1 : i-1] + (1.0f-f)*in[i]);
```

Comment:
Seems all very sensible - there was clearly an issue with the indexing here. But I'm struggling a bit to see what was causing the obvious failure in the arc case, which looks like it's using the wrong cut-plane normals - I can't see any of the current changes impacting on that... (?)

Reply:
I borked the arc sampling all the way back in #555. Arc initialisation was completely ignoring the midpoint provided by the user. :-/